### DISCUSSION TITLE
questions about gain code stability and metaflac

### DISCUSSION THREAD

**Discussion by atomGit** [complexlogic/rsgain#52] [Q&A] [answered]
hello @complexlogic 

1. i can't help but notice you're pumping out a lot of updates - my question is whether the actual gain code/algorithm is stable as far as the RG values that get written to the files, or whether it might change

2. regarding the consistency of the volume on a track by track basis (vs. album),  i'd like to hear any comments you might have with rsgain vs. metaflac

**Comment by complexlogic** [comment, marked as answer]
1. Yes, all the gain and peak values are the same in the new version. Note that this project uses semantic versioning `MAJOR.MINOR.PATCH`, where major increments include breaking changes, minor increments include new features but are fully backwards compatible with previous versions, and patch increments for bugfixes only. Currently, I don't plan any more major changes for this project, so the major version will remain at 3 unless something unforeseeable comes up.
2. Metaflac uses the old RG 1.0 algorithm, whereas rsgain uses the new RG 2.0 algorithm as defined in ITU-R BS.1770. Accordingly, you will notice different gain values between rsgain and metaflac. I would definitely recommend rsgain because the RG 2.0 algorithm is superior. I wrote some documentation on the differences between RG 1.0 and RG 2.0 [here](https://github.com/complexlogic/rsgain#normalizing-loudness).

**Reply by atomGit** [reply]
fantastic!
thanks much for the detailed answers

**Comment by atomGit** [comment]
fantastic! 
thanks for the detailed answers
